### PR TITLE
Update to detekt 1.17.1, Gradle 7.0.2 and IntelliJ 2021.1

### LINKED ISSUE #95: Find replacements for deprecated methods when increasing the minimum supported idea version to 2020.1-2
2020-07-19-161451_1309x615_scrot

### PR DESCRIPTION
**Pull request by osipxd**
Also, I've enabled plugin compatibility checks (`runPluginVerifier`) and removed the usage of deprecated APIs for annotation creation.

Closes #95

### CODE REVIEW

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
Looks good to me, though I have little knowledge on the IntelliJ plugin. Waiting for another review

File: build.gradle.kts
Comment (chao2zhang) [inline build.gradle.kts:13]: Nice!

**Review by chao2zhang — APPROVED** [review]
Thank you so much for taking care of this. I will do manual testing before publishing.

**Comment by chao2zhang** [comment]
Verified that this is working for me through `./gradlew runIde`

**Pull request merged**
